primitives/core/src/traits.rs

```diff
@@ -69,12 +85,12 @@ pub trait BareCryptoStore: Send + Sync {
 	/// Find intersection between provided keys and supported keys
 	///
 	/// Provided a list of (CryptoTypeId,[u8]) pairs, this would return
-	/// a filtered list of public keys which are supported by the keystore.
-	fn supported_keys(&self, id: KeyTypeId, keys: Vec<CryptoTypePublicPair>) -> Result<Vec<CryptoTypePublicPair>, String>;
+	/// a filtered set of public keys which are supported by the keystore.
+	fn supported_keys(&self, id: KeyTypeId, keys: Vec<CryptoTypePublicPair>) -> Result<HashSet<CryptoTypePublicPair>, Error>;
 	/// List all supported keys
 	///
-	/// Get a list of public keys the signer supports.
-	fn keys(&self, id: KeyTypeId) -> Result<Vec<CryptoTypePublicPair>, String>;
+	/// Returns a set of public keys the signer supports.
+	fn keys(&self, id: KeyTypeId) -> Result<HashSet<CryptoTypePublicPair>, Error>;
```

Comment:
(re my earlier comments, this would still return the entire type I think, because they are not bound to an incoming lifetime..?!?)